AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/edit-work-hours/vho-work-hours-non-availability-table/vho-work-hours-non-availability-table.component.ts

```diff
@@ -44,6 +44,7 @@ export class VhoWorkHoursNonAvailabilityTableComponent implements OnInit, CanDea
         }
         this.hideMessage();
         if (value) {
+           value = this.filterByFutureDate(value);
```

Comment:
I think that instead of passing all the data to the table component, just to filter it out, it would be better to filter it early and only pass data to the table that it should actually display. Maybe keep this table as a simple, presentation component.

Reply:
I would have to agree here too. For a table component that I would expect to traditionally be a presentation component, it's doing a fair bit. If possible, I would suggest you filter the data before passing it into the table component.

Reply:
I suggested Tahir do it here as the search component also fetches data for other tables, and as the filter is only required on this specific table, thought the responsibility should live there. Understand your reasonings however, makes sense, so I'm not too bothered when the logic sits.

Reply:
Addressed the issue mentioned in comment .

Reply:
I just realised the result field is used by two different components which both accept a different type. Ideally they should be two separate fields but that might have to be saved for another day

Reply:
agreed shaed